src/main/java/com/google/devtools/build/lib/sandbox/DockerSandboxedSpawnRunner.java

```diff
@@ -315,9 +314,11 @@ private String getOrCreateCustomizedImage(String baseImage) {
                         + "\"%d\", \"bazelbuild\"]\n",
                     gid, workDir, uid));
           }
-          dockerfile.append(
-              String.format("RUN [\"chown\", \"-R\", \"%d:%d\", \"%s\"]\n", uid, gid, workDir));
-          dockerfile.append(String.format("USER %d:%d\n", uid, gid));
+          if (uid > 0 || gid > 0) {
```

Comment:
This is a reasonable fix, but OTOH there's not much point left in creating a customized image if uid < 0 and gid < 0. We set the WORKDIR anyway on the `docker run` command line (https://source.bazel.build/bazel/+/master:src/main/java/com/google/devtools/build/lib/sandbox/DockerCommandLineBuilder.java;l=139) and by mounting a volume on that path, it is automatically created, making the "mkdir" call above useless, too.

I think a simpler fix might be to add a check above the call to computeIfAbsent (near https://source.bazel.build/bazel/+/master:src/main/java/com/google/devtools/build/lib/sandbox/DockerSandboxedSpawnRunner.java;l=291?q=DockerSandbox):

```diff
+    // We only need to create a customized image, if we're running on Linux, as Docker on macOS
+    // and Windows doesn't map users from the host into the container anyway.
+    if (OS.getCurrent() != OS.LINUX) {
+      return baseImage;
+    }
```

Reply:
Yes, this makes sense. I did not give too much thought to these changes, just hacked something together to make it work.